server/src/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1113,8 +1114,17 @@ public void cleanupStorage(boolean recurring) {
                     cleanupSecondaryStorage(recurring);
 
                     List<VolumeVO> vols = _volsDao.listVolumesToBeDestroyed(new Date(System.currentTimeMillis() - ((long) StorageCleanupDelay.value() << 10)));
+
                     for (VolumeVO vol : vols) {
                         try {
+                            // If this fails, just log a warning. It's ideal if we clean up the host-side clustered file
+                            // system, but not necessary.
+                            handleManagedStorage(vol);
+                        } catch (Exception e) {
```

Comment:
While I'd like this code to succeed (and automatically remove the relevant SR), if something does happen, I don't want it to interfere with normal operations (a manual cleanup of the SR we failed to remove can be done).

On Sep 14, 2016, at 10:08 AM, John Burwell <notifications@github.com<mailto:notifications@github.com>> wrote:


In server/src/com/cloud/storage/StorageManagerImpl.java<https://github.com/apache/cloudstack/pull/1642#discussion_r78780699>:

>                      for (VolumeVO vol : vols) {
>                          try {
> +                            // If this fails, just log a warning. It's ideal if we clean up the host-side clustered file
> +                            // system, but not necessary.
> +                            handleManagedStorage(vol);
> +                        } catch (Exception e) {


Why are all checked and unchecked exceptions being caught here?

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/apache/cloudstack/pull/1642/files/d3bdd6196e3005efdd29b318c84888d8098999ee#r78780699>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AC4SHyOdz_KUrC-CM97Ii_Vj-RiX0mOBks5qqBwGgaJpZM4Jlzqr>.